packages/local_auth/lib/local_auth.dart

```diff
@@ -92,9 +92,77 @@ class LocalAuthentication {
               'operating systems.',
           details: 'Your operating system is ${_platform.operatingSystem}');
     }
-    final bool? result =
-        await _channel.invokeMethod<bool>('authenticateWithBiometrics', args);
-    return result!;
+    return await _channel.invokeMethod<bool>(
+      'authenticateWithBiometrics',
+      args,
+    );
+  }
+
+  /// Authenticates the user with biometrics available on the device while also
+  /// allowing the user to use device authentication - pin, pattern, passcode.
+  ///
+  /// Returns a [Future] holding true, if the user successfully authenticated,
+  /// false otherwise.
+  ///
+  /// [localizedReason] is the message to show to user while prompting them
+  /// for authentication. This is typically along the lines of: 'Please scan
+  /// your finger to access MyApp.'
+  ///
+  /// [useErrorDialogs] = true means the system will attempt to handle user
+  /// fixable issues encountered while authenticating. For instance, if
+  /// fingerprint reader exists on the phone but there's no fingerprint
+  /// registered, the plugin will attempt to take the user to settings to add
+  /// one. Anything that is not user fixable, such as no biometric sensor on
+  /// device, will be returned as a [PlatformException].
+  ///
+  /// [stickyAuth] is used when the application goes into background for any
+  /// reason while the authentication is in progress. Due to security reasons,
+  /// the authentication has to be stopped at that time. If stickyAuth is set
+  /// to true, authentication resumes when the app is resumed. If it is set to
+  /// false (default), then as soon as app is paused a failure message is sent
+  /// back to Dart and it is up to the client app to restart authentication or
+  /// do something else.
+  ///
+  /// Construct [AndroidAuthStrings] and [IOSAuthStrings] if you want to
+  /// customize messages in the dialogs.
+  ///
+  /// Setting [sensitiveTransaction] to true enables platform specific
+  /// precautions. For instance, on face unlock, Android opens a confirmation
+  /// dialog after the face is recognized to make sure the user meant to unlock
+  /// their phone.
+  ///
+  /// Throws an [PlatformException] if there were technical problems with local
+  /// authentication (e.g. lack of relevant hardware). This might throw
+  /// [PlatformException] with error code [otherOperatingSystem] on the iOS
+  /// simulator.
+  Future<bool?> authenticate({
```

Comment:
Since you are making a breaking change already due to message strings, why not rename the method `authenticateWithBiometrics` as `authenticate` instead of having a near duplicate method?

That would leave the code much simpler.

Reply:
I wanted to keep the original behavior intact of using biometrics only with `authenticateWithBiometrics` and have `authenticate` use any means available, then perhaps a `authenticateWithCredentials` at some point.

If we remove `authenticateWithBiometrics`, I would want to add a way to keep the original biometric only behavior. Maybe a `bool biometricOnly = false,` parameter on `authenticate`?

Reply:
Yes, a `biometricOnly` parameter sounds good.

I like your original idea  as well (having separate methods) however the implementation (and API) is largely shared between these methods and you would need to make sure a feature added to `authenticate` is also added to all of its 
 variants (`authenticateWithBiometrics` etc).

You can reduce the implementation of `authenticateWithBiometrics` to just calling `authenticate` with `biometricOnly` set to true. However, I would rather keep the API clean and simple.

Reply:
Added a `biometricOnly` parameter to `authenticate`. `authenticateWithBiometrics` now calls `authenticate` with `biometricOnly: true`